sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java

```diff
@@ -133,7 +141,12 @@ public void resetAll() {
   @PostMapping("/{routeId}/reset")
   @Operation(summary = "Reset route", description = "Reset route")
   public void resetStatistics(@RouteIdParameter @PathVariable String routeId) {
-    mbeanContext.getManagedRoute(routeId).reset();
+    ManagedRouteMBean routeMBean = mbeanContext.getManagedRoute(routeId);
+    if (routeMBean == null) {
```

Comment:
can we reuse this validation block from above?

Reply:
Take a look at the new version. I've wrapped that in a function and replaced all occurrences of mbeanContext.getManagedRoute through the new function. The question is if 404 makes sense for other functions. It is hardly possible to get a null on something like this one:
camelContext.getRoutes().forEach(route -> getRouteMBean(route.getRouteId()).reset());
But take a look and see if it makes sense.